### PR TITLE
refactor PyTorchEngine check env

### PR DESCRIPTION
**Pull request by grimoire**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily receiving feedbacks. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Please describe the motivation of this PR and the goal you want to achieve through this PR.

## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repositories?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit tests to ensure the correctness.
3. If the modification has a dependency on downstream projects of a newer version, this PR should be tested with all supported versions of downstream projects.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

**Comment by lvhan028** [comment]
pr_ete_test failed

File: lmdeploy/pytorch/engine/engine_checker.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+            self.log_and_exit(
+                mod_name='Engine',
+                message='max_batch_size should be'
+                f' greater than 0, but got {engine_config.max_batch_size}')
+
+        if engine_config.dtype not in ['auto', 'float16', 'bfloat16']:
+            self.log_and_exit(
+                mod_name='Engine',
+                message=(
+                    f'unsupported specified data type {engine_config.dtype}'))
```
Comment (AllentDan) [inline lmdeploy/pytorch/engine/engine_checker.py:83]: Are they duplicates of https://github.com/InternLM/lmdeploy/blob/main/lmdeploy/messages.py#L283?

**Review by AllentDan — APPROVED** [review]

**Comment by AllentDan** [comment]
pr_ete_test failed again

File: requirements/runtime.txt [hunk 2]
```diff
@@ -18,5 +18,5 @@ tiktoken
 torch<=2.4.0,>=2.0.0
```
Comment (lvhan028) [inline requirements/runtime.txt:18]: pls change the upper version of pytorch being 2.5.1

**Review by RunningLeon — APPROVED** [review]
LGTM

**Review by jinminxi104 — APPROVED** [review]

**Comment by jinminxi104** [comment]
LGTM

**Pull request merged**